Library/Formula/agda.rb

```diff
@@ -51,38 +58,61 @@ def install
     cabal_clean_lib
 
     if build.with? "stdlib"
-      resource("stdlib").stage prefix/"agda-stdlib"
+      resource("stdlib").stage lib
 
-      # install the standard library's helper tools
-      cd prefix/"agda-stdlib" do
+      # generate the standard library's bytecode
+      cd lib do
         cabal_sandbox do
           cabal_install "--only-dependencies"
-          cabal_install "--prefix=#{prefix/"agda-stdlib"}"
-          system prefix/"agda-stdlib"/"bin"/"GenerateEverything"
-        end
-        cabal_clean_lib
-      end
-
-      # install the standard library's FFI bindings for the MAlonzo backend
-      if build.with? "malonzo-ffi"
-        cd prefix/"agda-stdlib"/"ffi" do
-          cabal_install "--user"
+          cabal_install
+          system "GenerateEverything"
         end
       end
 
       # generate the standard library's documentation and vim highlighting files
-      cd prefix/"agda-stdlib" do
+      cd lib do
         system bin/"agda", "-i", ".", "-i", "src", "--html", "--vim", "README.agda"
       end
+
+      rm_rf [lib/".cabal", lib/"dist"]
     end
 
-    # byte-compile and install Agda's included emacs mode
+    # compile the included Emacs mode
     if build.with? "emacs"
-      system bin/"agda-mode", "setup"
       system bin/"agda-mode", "compile"
     end
   end
 
+  def caveats
+    s = ""
+
+    if build.with? "emacs"
+      s += <<-EOS.undent
+
+      To add the Agda Emacs mode to your ~/.emacs file, run:
+        agda-mode setup
+      EOS
+    end
```

Comment:
I would really rather leave this in.  As I understand, it _is_ Homebre policy not to make modifications to the contents of the user’s home directory; otherwise, this command would’ve been executed automatically, as indeed it was earlier part of the formula.  Once again, Agda is expected to be used via Emacs, as I describe in #45322.